pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/AggregateOperator.java

```diff
@@ -123,11 +138,14 @@ public String toExplainString() {
   @Override
   protected TransferableBlock getNextBlock() {
     try {
-      if (!_readyToConstruct) {
-        consumeInputBlocks();
+      if (!_readyToConstruct && !consumeInputBlocks()) {
```

Comment:
made a minor change here to continue to producing the block instead of returning a no-op block when we read EOS from the input source